08_Scaling_to_multiple_GPUs/reference_solution/run_torchrun.sh

```diff
@@ -0,0 +1,50 @@
+#!/bin/bash
+#SBATCH --account=project_465001063
+#SBATCH --partition=standard-g
+#SBATCH --nodes=1
+#SBATCH --gpus-per-node=8
+#SBATCH --ntasks-per-node=1    # we start a single torchrun process, which will take care of spawning more
+#SBATCH --cpus-per-task=56     # 7 cores per GPU
+#SBATCH --mem-per-gpu=60G
+#SBATCH --time=0:15:00
+
+
+# Set up the software environment
+# NOTE: these modules will be available from the LUMI system stack after July 2024 and the "module use" line will no longer work
+module purge
+module use /project/project_465001063/modules
+module load singularity-userfilesystems singularity-CPEbits
+
+CONTAINER=/scratch/project_465001063/containers/pytorch_transformers.sif
+
+# Some environment variables to set up cache directories
+SCRATCH="/scratch/${SLURM_JOB_ACCOUNT}"
+FLASH="/flash/${SLURM_JOB_ACCOUNT}"
+export TORCH_HOME=$SCRATCH/torch-cache
+export HF_HOME=$FLASH/hf-cache
+mkdir -p $TORCH_HOME $HF_HOME
+
+# Disable internal parallelism of huggingface's tokenizer since we
+# want to retain direct control of parallelism options.
+export TOKENIZERS_PARALLELISM=false
+
+# Path to where the trained model and logging data will go
+export OUTPUT_DIR=$SCRATCH/$USER/data/
+export LOGGING_DIR=$SCRATCH/$USER/runs/
+export MODEL_NAME=gpt-imdb-model-${SLURM_JOBID}
+
```

Comment:
I am not entirely sure, but I think you also need to set `MIOPEN_USER_DB_PATH` and `MIOPEN_CUSTOM_CACHE_DIR` to some /tmp based locatgion when using multiple processes to avoid a potential race condition problem when the MIOpen DB/Cache is on the Lustre file system.

Reply:
As far as I know that one is only really used for convolutional kernels, which we don't use in this example. At least I have never encountered problems with it here

Reply:
but I can add them if you think that would be better

Reply:
"Less is more", so don't include it unless it's necessary. I think we should keeps things as simple as possible. I just wasn't aware that this was only needed when certain MIOpen kernels are used.

Reply:
Maybe we can confirm this today with Samuel